compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -657,7 +657,7 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
         else if (tp.classSymbol.is(CaseClass))
         // use constructor syntax for case class
           showType(tp) + signature(tp).map(_ => "_").mkString("(", ", ", ")")
-        else if (signature(tp).nonEmpty)
+        else if (sym.is(Flags.Case) && signature(tp).nonEmpty)
```

Comment:
Isn't this condition a superset of the condition of the previous if (`if (tp.classSymbol.is(CaseClass))`) and so this branch is just dead code?